test/e2e/framework/util.go

```diff
@@ -604,7 +604,7 @@ func WaitForPodsRunningReady(c clientset.Interface, ns string, minPods, allowedN
 				notReady++
 				badPods = append(badPods, pod)
 			default:
-				if _, ok := pod.Annotations[v1.CreatedByAnnotation]; !ok {
+				if controller.GetControllerOf(pod.GetObjectMeta()) == nil {
```

Comment:
`pod` is of type `"k8s.io/kubernetes/pkg/api/v1".Pod`, but `controller.GetControllerOf()` requires `k8s.io/apimachinery/pkg/apis/meta/v1".Object`.
`GetObjectMeta()` is to ensure the types match.